src/ddmd/console.d

```diff
@@ -8,116 +8,200 @@
+        {
+            enum FOREGROUND_WHITE = FOREGROUND_RED | FOREGROUND_GREEN | FOREGROUND_BLUE;
+            WORD attr = sbi.wAttributes;
+            attr = (attr & ~(FOREGROUND_WHITE | FOREGROUND_INTENSITY)) |
+                   ((color & COLOR.Red)   ? FOREGROUND_RED   : 0) |
+                   ((color & COLOR.Green) ? FOREGROUND_GREEN : 0) |
+                   ((color & COLOR.Blue)  ? FOREGROUND_BLUE  : 0) |
+                   (bright ? FOREGROUND_INTENSITY : 0);
+            SetConsoleTextAttribute(handle, attr);
+        }
+
+        /******************
+         * Reset console attributes to what they were
+         * when create() was called.
+         */
+        void resetColor()
+        {
+            SetConsoleTextAttribute(handle, sbi.wAttributes);
+        }
     }
     else version (Posix)
     {
-        const(char)* term = getenv("TERM");
-        return isatty(STDERR_FILENO) && term && term[0] && 0 != strcmp(term, "dumb");
-    }
-    else
-    {
-        return false;
-    }
-}
+        /* The ANSI escape codes are used.
+         * https://en.wikipedia.org/wiki/ANSI_escape_code
+         * Foreground colors: 30..37
+         * Background colors: 40..47
+         * Attributes:
+         *  0: reset all attributes
+         *  1: high intensity
+         *  2: low intensity
+         *  3: italic
+         *  4: single line underscore
+         *  5: slow blink
+         *  6: fast blink
+         *  7: reverse video
+         *  8: hidden
+         */
 
-extern (C++) void setConsoleColorBright(bool bright)
-{
-    version (Windows)
-    {
-        HANDLE h = GetStdHandle(STD_ERROR_HANDLE);
-        WORD attr = consoleAttributes(h);
-        SetConsoleTextAttribute(h, attr | (bright ? FOREGROUND_INTENSITY : 0));
-    }
-    else
-    {
-        fprintf(stderr, "\033[%dm", bright ? 1 : 0);
-    }
-}
+        import core.sys.posix.unistd;
 
-extern (C++) void setConsoleColor(COLOR color, bool bright)
-{
-    version (Windows)
-    {
-        HANDLE h = GetStdHandle(STD_ERROR_HANDLE);
-        WORD attr = consoleAttributes(h);
-        attr = (attr & ~(FOREGROUND_WHITE | FOREGROUND_INTENSITY)) | ((color & COLOR_RED) ? FOREGROUND_RED : 0) | ((color & COLOR_GREEN) ? FOREGROUND_GREEN : 0) | ((color & COLOR_BLUE) ? FOREGROUND_BLUE : 0) | (bright ? FOREGROUND_INTENSITY : 0);
-        SetConsoleTextAttribute(h, attr);
-    }
-    else
-    {
-        fprintf(stderr, "\033[%d;%dm", bright ? 1 : 0, 30 + cast(int)color);
-    }
-}
+      public:
+        FILE* fp;
 
-extern (C++) void resetConsoleColor()
-{
-    version (Windows)
-    {
-        HANDLE h = GetStdHandle(STD_ERROR_HANDLE);
-        SetConsoleTextAttribute(h, consoleAttributes(h));
+        static Console* create(FILE* fp)
+        {
+            import core.stdc.stdlib : getenv;
+            const(char)* term = getenv("TERM");
+            import core.stdc.string : strcmp;
+            if (!(isatty(STDERR_FILENO) && term && term[0] && 0 != strcmp(term, "dumb")))
```

Comment:
readably ?
`literal != functioncall` is wired. And not use in the rest of the codebase.
`mars.d` uses `!strcmp` throughout.